sqlite.go

```diff
@@ -556,3 +562,14 @@ func (err Error) Error() string {
 	}
 	return b.String()
 }
+
+// WithPersist makes a ctx instruct the sqlite driver to persist a prepared query.
```

Comment:
It's unfortunate to have to make a new context for this, because it allocates. It forces callers to allocate each time or maintain a separate "first-time-through" code path. I don't have any answers, just wanted to call it out.

Reply:
There are so many allocations through the use of database/sql it gets drowned in the noise. :-/

FWIW, a really enthusiastic user could `sqlite.WithPersist` all their queries once, and then not do that in the future. Though that would force a lot of the accounting work on the user I'm trying to avoid. I'll ponder workarounds. One would be a `func DefaultPersist(conn)` but that's surprisingly hard to use with database/sql's pool.